File: collector/src/main/java/com/bitmovin/analytics/enums/PlayerType.java
package com.bitmovin.analytics.enums;

import android.os.Parcel;
import android.os.Parcelable;

public enum PlayerType implements Parcelable {
    BITMOVIN("bitmovin"),
    EXOPLAYER("exoplayer"),
    AMAZON_IVS("amazonivs"); // TODO: check uppercase vs lowercase (we use Android:Exoplayer for PLAYER_TECH on the others)

    private final String name;

    PlayerType(String name) {
        this.name = name;
    }

    public static final Creator<PlayerType> CREATOR =
            new Creator<PlayerType>() {
                @Override
                public PlayerType createFromParcel(Parcel in) {
                    return valueOf(in.readString());
                }

                @Override
                public PlayerType[] newArray(int size) {
                    return new PlayerType[size];
                }
            };

    public String toString() {
        return this.name;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(name);
    }
}

File: collector/src/main/java/com/bitmovin/analytics/enums/PlayerType.kt
package com.bitmovin.analytics.enums

import android.os.Parcel
import android.os.Parcelable

enum class PlayerType(private val value: String) : Parcelable {
    BITMOVIN("bitmovin"),
    EXOPLAYER("exoplayer"),
    AMAZON_IVS("amazonivs"),
    MEDIA3_EXOPLAYER("media3-exoplayer"),
    THEOPLAYER("theoplayer"),
    ;

    override fun toString(): String {
        return value
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(
        parcel: Parcel,
        i: Int,
    ) {
        parcel.writeString(value)
    }

    companion object {
        @JvmField
        val CREATOR =
            object : Parcelable.Creator<PlayerType> {
                override fun createFromParcel(parcel: Parcel): PlayerType {
                    // default to bitmovin to be on the save side, otherwise we have to use '!!' to convert String? to String
                    return valueOf(parcel.readString() ?: BITMOVIN.toString())
                }

                override fun newArray(size: Int) = arrayOfNulls<PlayerType>(size)
            }
    }
}
